packages/react/src/components/select/custom/custom-option.tsx

```diff
@@ -0,0 +1,75 @@
+import { c, classy, m } from '@onfido/castor';
+import React, { MouseEvent, useEffect, useMemo, useRef } from 'react';
+import { useCustomSelect } from './useCustomSelect';
+
+let idCount = 0;
+
+export const CustomOption = ({
+  value: externalValue,
+  children: title,
+  className,
+  onMouseEnter,
+  onClick,
+  ...restProps
+}: CustomOptionProps): JSX.Element => {
+  const ref = useRef<HTMLButtonElement | null>(null);
+  const internalId = useMemo<number>(() => ++idCount, []);
+  const {
+    value: currentValue,
+    focusOption: currentFocusOption,
+    setValue,
+    setFocusOption,
+    addOption,
+    changeOption,
+    removeOption,
+  } = useCustomSelect();
+
+  const value = externalValue.toString();
+  const focused = currentFocusOption?.id === internalId;
+  const selected = currentValue === value;
+
+  useEffect(() => {
+    addOption(internalId, { value, title });
```

Comment:
Since the `ref` is already kept, the key could be that object's hash instead of a custom id we keep track of.

Reply:
I tried this, and it would work.

However, because I also use that ID as `key` when iterating over options and building `<option>`, I cannot use `ref` for that...

Reply:
I see. That's fine then, not a big deal.